### ISSUE TITLE
moor_generator >=2.0.1 is incompatible with json_serializable ^3.2.5

### ISSUE THREAD

**Issue by RakaAlrian** [simolus3/moor#364]
Please help, I can't start using moor because I got this error when I added moor_generator to pubspec.yaml. thanks

> Running "flutter pub get" in do_not...                          
Because analyzer_plugin >=0.2.1 depends on analyzer >=0.35.3 <0.39.0 and analyzer_plugin >=0.1.0 <0.2.1 depends on analyzer >=0.35.3 <0.38.0, analyzer_plugin >=0.1.0 requires analyzer >=0.35.3 <0.39.0.
And because moor_generator >=2.0.1 depends on analyzer_plugin >=0.1.0 <0.3.0, moor_generator >=2.0.1 requires analyzer >=0.35.3 <0.39.0.
And because json_serializable 3.2.5 depends on analyzer ^0.39.0 and no versions of json_serializable match >3.2.5 <4.0.0, moor_generator >=2.0.1 is incompatible with json_serializable ^3.2.5.
So, because do_not depends on both json_serializable ^3.2.5 and moor_generator ^2.3.1, version solving failed.
pub get failed (1; So, because do_not depends on both json_serializable ^3.2.5 and moor_generator ^2.3.1, version solving failed.)

**Comment by simolus3** [comment]
Thanks for the report! The next `moor_generator` version will support the latest analyzer and will therefore be compatible with the latest version of other builders. While the release needs some more work in the `moor` package, the next generator is stable enough to be used.

So you can work around this issue by putting this in your `pubspec.yaml`
```yaml
dependency_overrides:
  moor_generator:
    git:
      url: git@github.com:simolus3/moor.git
      ref: beta
      path: moor_generator
  sqlparser:
    git:
      url: git@github.com:simolus3/moor.git
      ref: beta
      path: sqlparser
```
Sorry for this inconvenience - I'll ping you after the next release when you can remove this.

**Comment by RakaAlrian** [comment]
Thank you for fast response, I tried to add those lines to my pubspec.yaml, but now I got this error

> Git error. Command: git clone --mirror git@github.com:simolus3/moor.git C:\FlutterSdk\.pub-cache\git\cache\moor-f3c1a9b90345ae935196e3531c15f1a51bfef385
Cloning into bare repository 'C:\FlutterSdk\.pub-cache\git\cache\moor-f3c1a9b90345ae935196e3531c15f1a51bfef385'...
Host key verification failed.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

**Comment by simolus3** [comment]
Ah, sorry. I think the https url should work
```yaml
dependency_overrides:
  moor_generator:
    git:
      url: https://github.com/simolus3/moor.git
      ref: beta
      path: moor_generator
  sqlparser:
    git:
      url: https://github.com/simolus3/moor.git
      ref: beta
      path: sqlparser
```

**Comment by RakaAlrian** [comment]
Well, at least I got no error for now. Thanks :D

**Comment by 2math** [comment]
You can use   json_serializable: ^3.2.3 until next version if is not important to use 3.2.5

**Comment by simolus3** [comment]
I just released `moor` and `moor_generator` version 2.4.0, which support the latest analyzer. You can remove the `dependency_overrides` section and just depend on `^2.4.0`. Let me know if you run into any problems, thanks!
